### DISCUSSION TITLE
in playlist repeat tracks very often with liquidsoap

### DISCUSSION THREAD

**Discussion by Ferbebu** [savonet/liquidsoap#3037] [User Support]
**Hello!** 

thanks I am using centovacast with liquidsoap, for radio it is repeating tracks very frequently, in play list, I want to know with what configuration to change this and be able to establish a minimum time before a track plays again, for example in Azuracast I can configure that they elapse 180 minutes before repeating a track or song.

 **Thanks for your help!**

**Comment by toots** [comment]
Hi @Ferbebu Could you share more of your script? This would help to find out how to best help you.

**Reply by Ferbebu** [reply]
thanks, I am using centovacast with liquidsoap, for radio, it is repeating tracks very frequently, in random mode, I want to know with what configuration to change this and to be able to establish a minimum time before a track sounds again, for example in Azuracast I can configure that 180 minuts elapse minutes before repeating a track or song. I require this configuration for radio in centovacast using liquidsoap thanks

Thanks for your help!

**Reply by Ferbebu** [reply]
# This is the settings configuration file for Centova Cast's LiquidSoap integration.
#
# Centova Cast will automatically update this file with the settings for the
# stream, but will try to preserve custom set() directives and changes to the
# centovacast.settings list.  In general it's not a good idea to modify this
# file unless strictly necessary.

# LiquidSoap configuration settings
set("log.file.path", "/usr/local/centovacast/var/vhosts/edwinperez/var/log/source.log")
set("server.socket.path", "/usr/local/centovacast/var/vhosts/edwinperez/var/run/liquidsoap.sock")
set("scheduler.fast_queues", 1)
set("server.telnet", false)
set("harbor.bind_addr", "0.0.0.0")
set("harbor.timeout", 30.0)
set("harbor.reverse_dns", false)


# Output configuration
centovacast.outputs = ref [
	("stream", output.shoutcast2(%mp3.cbr(stereo=true,stereo_mode="default",samplerate=44100,bitrate=128),host="82.212.95.12",port=9110,password="757iYg7o$CwERt",fallible=true,genre="No especificado",name="Experimental RADIO ONLINE",public=false,id="output_stream",url="http://radios.streaming.com/Experimental_RADIO",sid=1,portbase=9110,pwsid=true,adminpassword="RGVtb3JhZGlvQDQ3EF=="))
]
centovacast.outputs_mono = ref [

]


# Centova Cast integration configuration
centovacast.settings = [
	("username", "edwinperez"),
	("nextsong_uri", "/nextsong.php?username=edwinperez&secret=0f1c6935c3bacfbebba9f2629e10ac50"),
	("nextsong_debug", "0"),
	("fallback_playlist", "/usr/local/centovacast/var/vhosts/edwinperez/var/spool/playlist.txt"),
	("account_failsafe_filename", "/usr/local/centovacast/var/vhosts/edwinperez/var/spool/sounds/fallbackfile.mp3"),
	("system_failsafe_filename", "/usr/local/centovacast/var/spool/sounds/station-unavailable/general-fallback.mp3"),
	("autodj_startup_silence", "5"),
	("genre", ""),
	("crossfade", "1"),
	("crossfade_fadein", "0.1"),
	("crossfade_fadeout", "0.2"),
	("crossfade_mode", "normal"),
	("replay_gain", "1"),
	("portbase", "9112"),
	("harbor_password", "757iYg7o$CwPS"),
	("live_to_autodj_skip", "0"),
	("conservative", "0")
]

**Comment by toots** [comment]
Hi! I'm not sure if we can help. We provide the backend to azuracast but how azuracast uses liquidsoap is out of our hands.

**Reply by Ferbebu** [reply]
thanks my query is not for azuracast it is just an example that I have seen where if possible, I require the configuration for centovacast liquidsoap, I want to know with what configuration I can establish a minimum time before a track plays again, for example, configure allow 180 minutes to elapse before repeating a song or song.

**Reply by toots** [reply]
Ok. The `playlist` operator has a `check_next` function that you can use to refuse a song if it's been played too recently: https://www.liquidsoap.info/doc-2.1.4/reference.html#playlist

**Reply by Ferbebu** [reply]
Thank you
Do you have an example for the time 180 minutes?
Or how could it be implemented?
Thank you
